proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,100 @@
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+This is a customer requested functionality as discussed in this issue: https://github.com/NuGet/Home/issues/6060. Adding the functionality will aid our customers to be able to use the command when writing scripts.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+Imagine you have your dotnet CLI open. You would like to get the list of all packages from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search --source <MySource>`. It will provide you with the list of all the packages in the source `<MySource>`.
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options 
+
+| Option | Function |
+|---------|:----------|
+| `--Source` | A list of packages to search |
+| `--ExactMatch` | Return exact matches only as a search result |
+| `--Verbose` | Displays a detailed ist of information for each package |
+| `--Prerelease` | Allow prerelease packages to be shown. |
+| `--IncludeDelisted` | Allow unlisted packages to be shown |
+| `--NonInteractive` | Do not prompt for user input or confirmations.|
+| `--Help` | Show command help and usage information |
+| `--Verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+| `--ConfigFile` | The NuGet configuration file. If not specified, the hierarchy of configuration files from the current directory will be used. |
+|||
+#### **Algorithm**
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--Source`, `Search term`, `--Prerelease`, `--IncludeDelisted` values.
+3. Based on `--Verbosity` value, provide with a list of packages with various verbosity
+* Verbosity
+    - **Quiet** : Each line would look as follows : 
+    
+            <Package Name> <Latest Package Version>
+    - **Normal** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                <Description of the Package(short form)>
+    - **Detailed** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                Deprecated : YES/NO | Vulnerable : YES/NO
+                <Description of the Package(Readme file)>
+                License URL : <URL>
+
+#### **New Option `ExactMatch`**
```

Comment:
ExactMatch doesn't seem relevant to a spec on proving a `nuget.exe list` equivalent in the `dotnet` CLI.